library/src/main/java/ch/decent/sdk/crypto/ECKeyPair.kt

```diff
@@ -222,6 +220,23 @@ class ECKeyPair {
       compEnc[0] = (if (yBit) 0x03 else 0x02).toByte()
       return curve.curve.decodePoint(compEnc)
     }
+
+    /*
+          https://github.com/steemit/steem/issues/1944
+          bool public_key::is_canonical( const compact_signature& c )
+          {
+            return !(c.data[1] & 0x80)
+            && !(c.data[1] == 0 && !(c.data[2] & 0x80))
+            && !(c.data[33] & 0x80)
+            && !(c.data[33] == 0 && !(c.data[34] & 0x80));
+          }
+        */
+    @JvmStatic
+    fun checkCanonicalSignature(sigData: ByteArray): Boolean =
+        sigData.map { it.toInt() and 0xFF }.let {
+          it[1] < 0x80 && !(it[1] == 0 && it[2] < 0x80)
+              && it[33] < 0x80 && !(it[33] == 0 && it[34] < 0x80)
```

Comment:
are you sure about this? This is just a bunch of comparisons... in the code in the comment c.data[2] for example is anded with 0x80, you're just comparing that c.data[2] and 0xFF is smaller. Which, looking at the linked issue is similar to the pseudocode described by the poster of the issue, but his suggested code seems to be different to what's going on here.

Reply:
yes
`!(c.data[1] & 0x80)` means eg. `100 & 128 = 0`, `128 & 128 = 128`, `150 & 128 = 128`
so smaller values then `0x80` are equal to zero, which equals to `false` and negated to `true`
so all numbers smaller then 0x80 should be `true`